Plugins/BTCPayServer.Plugins.ShopifyPlugin/Services/ShopifyHostedService.cs

```diff
@@ -103,7 +103,7 @@ async Task<InvoiceLogs> Process(long shopifyOrderId, InvoiceEntity invoice)
                 .Where(h => h is { Kind: "REFUND", Status: "SUCCESS" }).ToArray();
 
         bool canRefund = captures.Length > 0 && captures.Length > refunds.Length;
-        if (invoice.Status is InvoiceStatus.Settled)
+        if (invoice.Status is InvoiceStatus.Settled || (invoice.Status == InvoiceStatus.Expired && invoice.ExceptionStatus == InvoiceExceptionStatus.PaidPartial))
```

Comment:
Try to use pattern matching in C#, it increases readability:

```suggestion
        if (invoice is { Status: InvoiceStatus.Settled } or { Status:  InvoiceStatus.Expired, ExceptionStatus: InvoiceExceptionStatus.PaidPartial })
```